storage/tool/src/main/java/com/hortonworks/registries/storage/tool/sql/SchemaFlywayFactory.java

```diff
@@ -33,10 +36,62 @@
     private static final boolean cleanOnValidationError = false;
 
 
-    public static Flyway get(ClassLoader classLoader,
-                             StorageProviderConfiguration conf,
-                             String scriptRootPath,
-                             boolean validateOnMigrate) {
+    static Flyway get(ClassLoader classLoader,
+                      StorageProviderConfiguration conf,
+                      String scriptRootPath,
+                      boolean validateOnMigrate) {
+        switch (conf.getDbType()) {
+            case MYSQL:
+                return mysqlFlyway(classLoader, conf, scriptRootPath, validateOnMigrate);
+            case POSTGRESQL:
+                return postgresFlyway(classLoader, conf, scriptRootPath, validateOnMigrate);
+            case ORACLE:
+                return oracleFlyway(classLoader, conf, scriptRootPath, validateOnMigrate);
+            default:
+                throw new IllegalArgumentException("Unknown database : " + conf.getDbType());
+        }
+    }
+
+    private static Flyway mysqlFlyway(ClassLoader classLoader,
```

Comment:
Just a suggestion:
1. Both mysql and postgres can be handled in the switch case fall-through.
2. For oracle, I think it's enough to pass the `basicFlyway` and `StorageProviderConfiguration`.
3. No need to pass `null` for varargs.

Reply:
As far as 1 and 2 is concerned I expect MySQL and Postgresql specific properties to come in once we add TLS support for it as well. That is why I kept the mysql and postgresql configs in their own functions.